src/workloads/query/TelemetryWithRedaction.yml

```diff
@@ -0,0 +1,193 @@
+    Type: AdminCommand
+    Threads: 32
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "OnlyActiveInPhase", Default: 1024}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Duration: 1 minute
+          SleepBefore: 5 seconds
+          Database: admin
+          Operations:
+          - OperationMetricsName: TelemetryWithRedactionMetrics
+            OperationName: AdminCommand
+            OperationCommand:
+              aggregate: 1
+              pipeline: [
+                      {
+                        $telemetry: {
+                          redactIdentifiers: {^Parameter: {Name: "redactIdentifiers", Default: false}},
+                          redactionKey: *binDataGen
+                        }
+                      }
+                    ]
+              cursor: {}
+
+Actors:
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: InsertData
+  Type: Loader
+  Threads: 4
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document:
+          a: &integer {^RandomInt: {min: -100, max: 100}}
+          b: *integer
+          string: &string {^RandomString: {length: 5}}
+        Indexes:
+          - keys: {a: 1, b: 1}
+          - keys: {string: 1}
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+
+- Name: FindCommand
+  Type: CrudActor
+  Threads: 32
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [3]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: 1 minute
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {string: {$in: ["hello", "hi", "bye"]}}
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [4]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+
+- Name: FindCommandWithRedaction
+  Type: CrudActor
+  Threads: 32
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [5]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: 1 minute
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {string: {$in: ["hello", "hi", "bye"]}}
```

Comment:
So this will create just one query shape for us to consume in $telemetry, is that right? I think we should try to test with a bunch of different query shapes - which would probably be more realistic of the environment where we would worry about the performance of $telemetry. Could you borrow from the other workload you just did to make that happen?

I think also we'd probably want to try to vary the size of the query shapes if we could? Would it be possible to have the queries be small-ish (query on a couple fields, fits roughly on 1 or 2 lines if you have it all in json) to large-ish (hundreds of fields, ideally approaching like 1 to 10 KB in bson size?)

Reply:
Uhh you can test it out in the shell depending on the format:
```js
Object.bsonsize({x: 1})
16
```

Reply:
Okay thanks!